### PR TITLE
Create SPACE Cancer Disease

### PR DESCRIPTION
**Pull request by Diegoflores31**
I want to add Space Cancer disease , If someone can check it it would be good because im new on this code thing

**What does this PR do:**
Adds SPACE cancer Disease , a terminal non contagius thing for the admemes to punish bad doers and smokers 

**Changelog:**
:cl:
add: Cancer Disease
/:cl:

### CODE REVIEW

File: code/datums/diseases/Cancer [hunk 1]
```diff
@@ -0,0 +1,45 @@
+/datum/disease/Cancer
+	form = "Condition"
+	name = "Cancer"
+	max_stages = 5
+	cure_text = "Surgery"
+	agent = "Cancer Cells"
+	viable_mobtypes = list(/mob/living/carbon/human)
+	permeability_mod = 1
+	desc = "If left untreated the subject will become very weak,Massive Organ Failure and Die."
```
Comment (farie82) [inline code/datums/diseases/Cancer]: grammar in this sentence.
If left untreated the subject will become very weak, have an organ failure and die.
Something the like

**Comment by Superhats** [comment]
That's a big no for me. Looking at it, you'll need to remove the patient's lungs to cure them, it's filled with grammatical errors, and I believe instantly throwing 280 damage on someone at stage 5 isn't how viruses should do their damage.

**Comment by Diegoflores31** [comment]
thats how cancer works

File: code/datums/diseases/Cancer [hunk 2]
```diff
@@ -0,0 +1,45 @@
...
+			if(A)
+				A.inflamed = 1
+				A.update_icon()
+			if(prob(3))
+				to_chat(affected_mob, "<span class='warning'>You feel Pain in your chest!</span>")
+				affected_mob.adjustOxyLoss(10)
+		if(3)
+			if(prob(1))
+				to_chat(affected_mob, "<span class='warning'>You cant Breathe Normaly!</span>")
+    if(4)
```
Comment (farie82) [inline code/datums/diseases/Cancer]: misses a tab from here to the end. It is not part of the switch statement anymore from here

File: code/datums/diseases/Cancer [hunk 3]
```diff
@@ -0,0 +1,45 @@
...
+		if(1)
+			if(prob(5))
+				affected_mob.emote("cough")
+		if(2)
+			var/obj/item/organ/internal/Lungs/A = affected_mob.get_int_organ(/obj/item/organ/internal/Lungs)
+			if(A)
+				A.inflamed = 1
+				A.update_icon()
+			if(prob(3))
+				to_chat(affected_mob, "<span class='warning'>You feel Pain in your chest!</span>")
```
Comment (farie82) [inline code/datums/diseases/Cancer]: You feel a sharp pain in your chest!
Or something the like.

**Comment by AffectedArc07** [comment]
Few things of note:
1: This appears to have been made with the web editor, so that is gonna lessen your merge ability because testing probably didnt go into this
2: Your description is lackluster
3: ***WHY***

**Comment by datlo** [comment]
Please polish and test your PR before submitting it. Right now I don't even think the disease would ever spawn on anybody in the first place.

Also, cancer doesn't instantly kills you on the spot in a less than 2-hour frame after you contract it.

File: code/datums/diseases/Cancer [hunk 4]
```diff
@@ -0,0 +1,45 @@
...
+			if(prob(1))
+				to_chat(affected_mob, "<span class='warning'>You cant Breathe Normaly!</span>")
+    if(4)
+      if(prob(1))
+        to_chat(affected_mob, "<span class='warning'>You feel the pain expanding to all your body!</span>")
+        affected_mob.AdjustWeaken(10)
+        affected_mob.AdjustToxinloss(30)
+    if(5)
+      if(prob(4)
+        to_chat(affected_mob, "<span class='warning'>You cant Figth it anymore!</span>")
```
Comment (farie82) [inline code/datums/diseases/Cancer]: "You can\'t fight it anymore!"
or something

**Comment by Diegoflores31** [comment]
its not supposed to spawn on someone yet , just an admeme disease

**Comment by AffectedArc07** [comment]
@Diegoflores31 its not a DM file, so its not even "in the code". This means that it does **NOTHING** inside the game code.

**Comment by Spacemanspark** [comment]
What the fuck

**Comment by lass9030** [comment]
Please, download the files and use an editor like Visual Studio Code, this just hurts to look at. Help is available in #coder_chat on discord.

**Comment by AffectedArc07** [comment]
image

**Comment by Mitchs98** [comment]
I...just no. It's the future. There's a long running meme of cancer being cured, I'm pretty sure they'd of cured it already anyways. If we can cure a cold and diseases that make you spit up dogs....lung cancer unless it's some form of hyper cancer 2.0 would be childs play.

**Comment by Diegoflores31** [comment]
Cancer Update , grammar Fixed and the things that you told me too

**Comment by Ty-Omaha** [comment]
This still doesn't add the file to the dme.

Seriously - please test your PR.

**Comment by datlo** [comment]
I don't want to be rude but please just close this and come talk in the coder chat discord and learn how to do things properly, plenty of people are willing to help a new coder if you're willing to listen.

**Comment by Spartan6** [comment]
Uhm.

**Comment by TullyBurnalot** [comment]
... dunno why I'm doing this, but sure, why not?

1) Cancer is not a virus, airborne disease, or anything remotely related to the way Paradise currently handles diseases from a mechanical standpoint. Contagious viral agents that make your body explode are stupid, but they're at least an actual virus. Cancer is not a virus. It does not "spread" like a virus does in our current disease mechanics;

2) The desc reads, and I quote "The Old and Trusty Cancer but in SPACE". I am fully aware that this game is chock-full of black humour, but this is just extremely badly delivered and in poor taste;

3) Cancer does not work by randomly making you choke yourself into critical condition. That is not how "cancer works";

4) Even after the "fix", the descriptions are full of poor syntax, grammar, and following in Point 2, such poorly tasteful things as "You feel CANCEROUS!".  If your intention was to make it sound serious, I'm not sure you succeeded;

5) Refer to the myriad of technical issues reported above.

While it's great that you want to contribute to the codebase, it would be far better if you started off small and took plenty of advice from folks over in the public coder chat, because this PR is... not really close to being in an acceptable state from any perspective.

**Comment by Fox-McCloud** [comment]
I'm really not sure how this could remotely be made interesting. This was tried on VG, and it ended up just being an irritating feature of medical--you ended up having to have special drugs applied, repeatedly, that kept you poisoned and bedridden, you had to stay in one spot in medbay, pretty much, and it just generally took up a ton of time.

While some aspects of "random infectious disease that take up time" are interesting or have some justification to exist--this niche has already kinda been filled with such things like the bad appendix event, the cold, flu, or what have you.

While the disease system is easily tailored to allowing for non-viral/bacterial disease, with purely a few flavortexts, mechanically, I don't see how this could be all that interesting, mechanically.

File: code/datums/diseases/Cancer.dm [hunk 5]
```diff
@@ -0,0 +1,61 @@
...
+/datum/disease/Cancer/stage_act()
+	..()
+	switch(stage)
+		if(1)
+			if(prob(5))
+				affected_mob.emote("cough")
+		if(2)
+			var/obj/item/organ/internal/Lungs/A = affected_mob.get_int_organ(/obj/item/organ/internal/Lungs)
+			if(A)
+				A.inflamed = 1
```
Comment (Fox-McCloud) [inline code/datums/diseases/Cancer.dm]: so you literally copy-pasted appendix code, then didn't include it in the .dme

Yeah, no, this is not going to pass--include it, and test your stuff, please.

**Pull request closed without merging**